File: judge-api/src/main/kotlin/io/leonfoliveira/judge/api/controller/dto/response/ProblemResponseDTO.kt
package io.leonfoliveira.judge.api.controller.dto.response

import io.leonfoliveira.judge.core.service.dto.output.ProblemOutputDTO
import java.util.UUID

data class ProblemResponseDTO(
    val id: Int,
    val title: String,
    val descriptionKey: UUID,
    val timeLimit: Int,
    val testCasesKey: UUID,
)

fun ProblemOutputDTO.toResponseDTO(): ProblemResponseDTO {
    return ProblemResponseDTO(
        id = id,
        title = title,
        descriptionKey = descriptionKey,
        timeLimit = timeLimit,
        testCasesKey = testCasesKey,
    )
}

File: judge-web/src/core/repository/dto/response/ProblemResponseDTO.ts
import { DownloadAttachmentResponseDTO } from "@/core/repository/dto/response/DownloadAttachmentResponseDTO";

export type ProblemResponseDTO = {
  id: number;
  title: string;
  description: string;
  timeLimit: number;
  testCases: DownloadAttachmentResponseDTO;
};
